modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/PathType.java

```diff
@@ -108,6 +108,27 @@ URL resolvePath(final String parentLocation, final String location) {
             }
         }
 
+    },
+    
+    CUSTOM {
```

Comment:
So "CUSTOM" actually does nothing, and should be used in case the path is already a URL? How about calling it "URL" instead, would seem clearer. The logging is also misleading imho, it talks about a parent location, but it's actually never used.